File: backend/src/main/java/com/team05/petmeeting/domain/user/dto/profile/UserFeedRes.java
package com.team05.petmeeting.domain.user.dto.profile;

import com.team05.petmeeting.domain.feed.entity.Feed;

import java.time.format.DateTimeFormatter;
import java.util.List;

public record UserFeedRes(
        long totalFeedCount,
        List<ProfileFeedItem> feeds
) {
    public static UserFeedRes of(long totalFeedCount, List<Feed> feedList) {
        List<ProfileFeedItem> items = feedList.stream()
                .map(ProfileFeedItem::from)
                .toList();
        return new UserFeedRes(totalFeedCount, items);
    }

    public record ProfileFeedItem(
            Long feedId,
            String category,
            String title,
            String createdAt
    ) {
        public static ProfileFeedItem from(Feed feed) {
            return new ProfileFeedItem(
                    feed.getId(),
                    feed.getCategory().toString(),
                    feed.getTitle(),
                    feed.getCreatedAt().format(DateTimeFormatter.ofPattern("yyyy. MM. dd."))
            );
        }
    }
}

File: backend/src/main/java/com/team05/petmeeting/domain/user/dto/profile/UserFeedRes.kt
package com.team05.petmeeting.domain.user.dto.profile

import com.team05.petmeeting.domain.feed.entity.Feed
import java.time.format.DateTimeFormatter

data class UserFeedRes(
    val totalFeedCount: Long,
    val feeds: MutableList<ProfileFeedItem>
) {

    data class ProfileFeedItem(
        val feedId: Long,
        val category: String,
        val title: String,
        val createdAt: String
    ) {
        companion object {
            @JvmStatic
            fun from(feed: Feed): ProfileFeedItem {
                return ProfileFeedItem(
                    feed.id,
                    feed.category.toString(),
                    feed.title,
                    feed.createdAt.format(DateTimeFormatter.ofPattern("yyyy. MM. dd."))
                )
            }
        }
    }

    companion object {
        @JvmStatic
        fun of(totalFeedCount: Long, feedList: MutableList<Feed>): UserFeedRes {
            val items = feedList.stream()
                .map { feed: Feed -> ProfileFeedItem.from(feed) }
                .toList()
            return UserFeedRes(totalFeedCount, items)
        }
    }
}